docs/docs/dev_docs/contracts/syntax/storage.md

```diff
@@ -1,40 +1,533 @@
+  - The term "some accumulation" is intentionally vague. The interpretation of the "current value" of a `Set` must be expressed by the smart contract developer. A common use case for a `Set` is to represent the sum of a collection of values (in which case 'accumulation' is 'summation').
+    - Think of a ZCash balance (or even a Bitcoin balance). The "current value" of a user's ZCash balance is the sum of all unspent (not-yet nullified) notes belonging to that user.
+  - To modify the "current value" of a `Set` state variable, is to [`insert`](#insert) new notes into the `Set`, or [`remove`](#remove) notes from that set.
+  - Interestingly, if a developer requires a private state to be modifiable by users who _aren't_ privy to the value of that state, a `Set` is a very useful type. The `insert` method allows new notes to be added to the `Set` without knowing any of the other notes in the set! (Like posting an envelope into a post box, you don't know what else is in there!).
+
+
+
+
+## `Singleton<NoteType>`
+
+Singleton is a private state variable that is unique in a way. When a Singleton is initialized, a note is created to represent its value. And the way to update the value is to destroy the current note, and create a new one with the updated value.
+
+Like for public state, we define the struct to have context, a storage slot and `note_interface` specifying how the note should be constructed and manipulated.
+
+#include_code struct /yarn-project/aztec-nr/aztec/src/state_vars/singleton.nr rust
+
+An example of singleton usage in in the account contracts keeping track of public keys. The `Singleton` is added to the `Storage` struct as follows:
+
+#include_code storage-singleton-declaration /yarn-project/noir-contracts/src/contracts/docs_example_contract/src/main.nr rust
+
+### `new`
+
+#include_code new /yarn-project/aztec-nr/aztec/src/state_vars/singleton.nr rust
+
+As part of the initialization of the `Storage` struct, the `Singleton` is created as follows, here at storage slot 1 and with the `NoteInterface` for `CardNote`.
+
+#include_code start_vars_singleton /yarn-project/noir-contracts/src/contracts/docs_example_contract/src/main.nr rust
+
+### `initialise`
+
+As mention, the singleton is initialized to create the first note and value. 
+
+#include_code initialise /yarn-project/aztec-nr/aztec/src/state_vars/singleton.nr rust
+
+When this function is called, a nullifier of the storage slot is created, preventing this Singleton from being initialized again.
+
+Unlike public states, which have a default initial value of `0` (or many zeros, in the case of a struct, array or map), a private state (of type `Singleton`, `ImmutableSingleton` or `Set`) does not have a default initial value. The `initialise` method (or `insert`, in the case of a `Set`) must be called.
+
+:::info
+Extend on what happens if you try to use non-initialized state.
```

Comment:
Is this a note to self?

Reply:
Partially to self or someone walking through to elaborate. I haven't tested the effects currently, but should have time to do it today and add the results in here.